File: StockChess/StockChess/Helpers/ViewModelLocator.vb
Imports Microsoft.Practices.Unity

Public Class ViewModelLocator
    Private _container As UnityContainer

    Public Sub New()
        _container = New UnityContainer
        _container.RegisterType(Of IEngineService, StockfishService)()
    End Sub

    Public ReadOnly Property ChessVM As ChessViewModel
        Get
            Return _container.Resolve(Of ChessViewModel)()
        End Get
    End Property
End Class

File: StockChess/StockChessCS/Helpers/ViewModelLocator.cs
using Microsoft.Practices.Unity;
using StockChessCS.Interfaces;
using StockChessCS.Services;
using StockChessCS.ViewModels;

namespace StockChessCS.Helpers
{
    public class ViewModelLocator
    {

        private UnityContainer _container;
        public ViewModelLocator()
        {
            _container = new UnityContainer();
            _container.RegisterType<IEngineService, StockfishService>();
        }

        public ChessViewModel ChessVM
        {
            get { return _container.Resolve<ChessViewModel>(); }
        }
    }
}
